src/GraphQL/QueryFieldConfigGenerator/CalculatedValue.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Enterprise License (PEL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ *  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ *  @license    http://www.pimcore.org/license     GPLv3 and PEL
+ */
+
+namespace Pimcore\Bundle\DataHubBundle\GraphQL\QueryFieldConfigGenerator;
+
+use Pimcore\Model\DataObject\ClassDefinition\Data;
+
+/**
+ * Class CalculatedValue
+ * @package Pimcore\Bundle\DataHubBundle\GraphQL\QueryFieldConfigGenerator
+ */
+class CalculatedValue extends Base
+{
+    /**
+     * @param $attribute
+     * @param Data $fieldDefinition
+     * @param null $class
+     * @param null $container
+     *
+     * @return mixed
+     */
+    public function getGraphQlFieldConfig($attribute, Data $fieldDefinition, $class = null, $container = null)
+    {
+        return $this->enrichConfig(
+            $fieldDefinition,
+            $class,
+            $attribute,
+            [
+                'name' => $fieldDefinition->getName(),
+                'type' => $this->getFieldType($fieldDefinition, $class, $container),
```

Comment:
which defaults to string, right?

Reply:
Yes, it inherits `getFieldType()` from Base, and as far as I understand it will always be a string, [even the numerical addition example in docs returns a string ](/pimcore/pimcore/blob/master/doc/Development_Documentation/05_Objects/01_Object_Classes/01_Data_Types/33_Calculated_Value_Type.md).

Reply:
True, which is kind of sad because the only reason for that is that it needs a string for displaying it in the grid + for csv export. Maybe this restriction should be relaxed.